### PR TITLE
Added chose category dropdown for new quiz

### PR DESCRIPTION
**Pull request by mynotdoing**
Added chose category dropdown for new quiz

Знімок екрана 2023-11-22 о 16 05 06

https://github.com/ita-social-projects/SpaceToStudy-Client/assets/75339353/187b75b1-bc21-4ec8-a2fc-e3b4637b4c15

### CODE REVIEW

**Review by MariaAfonina — COMMENTED** [review]
Please add a picture or video of how it works

File: src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.styles.ts [hunk 1]
```diff
@@ -62,5 +69,22 @@ export const styles = {
       gap: '12px',
       width: '100%'
     }
+  },
+  categoryInput: {
+    width: '45%',
```
Comment (Tolik170) [inline src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.styles.ts:74]: should it be 45%?

File: src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx [hunk 2]
```diff
@@ -31,6 +35,16 @@ const EditQuizContainer = ({ questions, setQuestions }: QuizContentProps) => {
   const [title, setTitle] = useState<string>('')
   const [description, setDescription] = useState<string>('')
   const [isCreationOpen, setIsCreationOpen] = useState<boolean>(false)
+  const [searchParams, setSearchParams] = useSearchParams()
```
Comment (Tolik170) [inline src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx:38]: you don`t need to make any manipulation with url, it should be state like with title and desription

File: src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx [hunk 3]
```diff
@@ -78,6 +92,18 @@ const EditQuizContainer = ({ questions, setQuestions }: QuizContentProps) => {
           value={description}
           variant={TextFieldVariantEnum.Standard}
         />
+        <AsyncAutocomplete
```
Comment (Tolik170) [inline src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx:95]: better pass fetchOnMount prop too

File: src/utils/helper-functions.tsx [hunk 4]
```diff
@@ -180,7 +180,7 @@ export const spliceSx = (
   ({
     ...defaultStyles,
     ...newStyles
-  } as SxProps)
+  }) as SxProps
```
Comment (Tolik170) [inline src/utils/helper-functions.tsx:183]: fix this error, looks like you have some troubles with local prettier and lint

File: src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx [hunk 5]
```diff
@@ -78,6 +92,18 @@ const EditQuizContainer = ({ questions, setQuestions }: QuizContentProps) => {
...
+          labelField='name'
+          onChange={onCategoryChange}
+          service={categoryService.getCategoriesNames}
+          sx={styles.categoryInput}
+          textFieldProps={{
+            label: t('myResourcesPage.quizzes.categoryDropdown'),
+            helperText: t('myResourcesPage.quizzes.categoryDropdownHelper')
```
Comment (Tolik170) [inline src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx:102]: I guess, you don`t need this helper text, category field is not required and you can only choose values from options list, so, validation is redundant here

**Comment by abalanovsky** [comment]
Fix lint please

File: src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.styles.ts [hunk 6]
```diff
@@ -62,5 +69,22 @@ export const styles = {
...
+    mt: '30px',
+    mb: '10px',
+    maxWidth: { md: '370px' },
+    mr: '30px',
```
[the remark this answers is not in the document]
Comment (mynotdoing) [reply src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.styles.ts:78]: done

File: src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx (hunk 2 above)
Comment (mynotdoing) [reply src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx:38]: done

File: src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx (hunk 5 above)
Comment (mynotdoing) [reply src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx:102]: but helperText only describes the category field, it doesn't do any validations. We need it as it is showed in figma design.

File: src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.styles.ts (hunk 1 above)
Comment (mynotdoing) [reply src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.styles.ts:74]: fixed

File: src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx (hunk 3 above)
Comment (mynotdoing) [reply src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx:95]: fixed

File: src/utils/helper-functions.tsx (hunk 4 above)
Comment (mynotdoing) [reply src/utils/helper-functions.tsx:183]: fixed

File: src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx [hunk 7]
```diff
@@ -30,8 +33,16 @@ const EditQuizContainer = ({ questions, setQuestions }: QuizContentProps) => {
...
   const [title, setTitle] = useState<string>('')
   const [description, setDescription] = useState<string>('')
+  const [category, setCategory] = useState<string>('')
   const [isCreationOpen, setIsCreationOpen] = useState<boolean>(false)
 
+  const onCategoryChange = (
+    _: React.SyntheticEvent,
+    value: CategoryNameInterface | null
+  ) => {
+    setCategory(value?.name ?? '')
```
Comment (Tolik170) [inline src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx:43]: value?._id

Comment (mynotdoing) [reply src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx:43]: fixed

File: src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx [hunk 8]
```diff
@@ -30,8 +33,16 @@ const EditQuizContainer = ({ questions, setQuestions }: QuizContentProps) => {
...
+  const [category, setCategory] = useState<string>('')
   const [isCreationOpen, setIsCreationOpen] = useState<boolean>(false)
 
+  const onCategoryChange = (
+    _: React.SyntheticEvent,
+    value: CategoryNameInterface | null
+  ) => {
+    setCategory(value?._id ?? '')
+  }
+
```
Comment (abalanovsky) [inline src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx:45]: Try to increase test coverage, at least on your new code
(You can see all coverage on Sonar, or locally at vitest report)

**Review by maarekua — APPROVED** [review]

File: src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.constants.ts [hunk 9]
```diff
@@ -4,7 +4,7 @@ export const defaultResponse = {
   description: '',
   items: [],
   author: { _id: '' },
-  category: null,
+  category: '',
```
Comment (Tolik170) [inline src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.constants.ts:7]: I think default value should be null

File: src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.tsx [hunk 10]
```diff
@@ -40,14 +45,23 @@ const CreateOrEditQuizContainer = ({
...
   const { setAlert } = useSnackBarContext()
   const { openModal } = useModalContext()
   const navigate = useNavigate()
   const [isCreationOpen, setIsCreationOpen] = useState<boolean>(false)
 
+  const onCategoryChange = (
+    _: React.SyntheticEvent,
+    value: Category | null
+  ) => {
+    setCategory(value?._id ?? '')
```
Comment (Tolik170) [inline src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.tsx:62]: setCategory(value?._id ?? null)

File: src/pages/new-quiz/NewQuiz.tsx [hunk 11]
```diff
@@ -12,6 +12,7 @@ const NewQuiz = () => {
   const [questions, setQuestions] = useState<Question[]>([])
   const [title, setTitle] = useState<string>('')
   const [description, setDescription] = useState<string>('')
+  const [category, setCategory] = useState<string>('')
```
Comment (Tolik170) [inline src/pages/new-quiz/NewQuiz.tsx:15]: const [category, setCategory] = useState<string | null>(null)

File: src/types/quizzes/interfaces/quizzes.interface.ts [hunk 12]
```diff
@@ -15,11 +9,12 @@ export interface Quiz extends CommonEntityFields {
...
   author: Pick<UserResponse, '_id'>
-  category: Category | null
+  category: string
 }
 
 export interface CreateQuizParams {
   title: string
   description: string
   items: Question[]
+  category: string
```
Comment (Tolik170) [inline src/types/quizzes/interfaces/quizzes.interface.ts:19]: category: string | null

File: src/pages/new-quiz/NewQuiz.tsx (hunk 11 above)
Comment (mynotdoing) [reply src/pages/new-quiz/NewQuiz.tsx:15]: fixed

File: src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.tsx (hunk 10 above)
Comment (mynotdoing) [reply src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.tsx:62]: fixed

File: src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.constants.ts (hunk 9 above)
Comment (mynotdoing) [reply src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.constants.ts:7]: fixed

File: src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx (hunk 8 above)
Comment (mynotdoing) [reply src/containers/my-quizzes/edit-quiz-container/EditQuizContainer.tsx]: fixed

File: src/constants/translations/en/my-resources-page.json [hunk 13]
```diff
@@ -46,7 +46,9 @@
     "createNewQuestion": "Create new question",
     "editQuestion": "Edit question",
     "addQuestion": "Add question",
-    "successAddedOuiz": "Ouiz was successtully created"
+    "successAddedOuiz": "Ouiz was successtully created",
+    "categoryDropdown": "Category...",
+    "categoryDropdownHelper": "Choose the category:"
```
[the remark this answers is not in the document]
Comment (mynotdoing) [reply src/constants/translations/en/my-resources-page.json:51]: fixed

File: src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.styles.ts [hunk 14]
```diff
@@ -12,6 +12,13 @@ const getLabelStyles = (value: string) => ({
   color: palette.primary[300]
 })
 
+const commonStyle = {
+  fontSize: '14px',
+  position: 'absolute',
+  top: '-33px',
+  left: '-14px'
+}
```
[the remark this answers is not in the document]
Comment (mynotdoing) [reply src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.styles.ts:20]: fixed

File: src/pages/new-quiz/NewQuiz.constants.tsx [hunk 15]
```diff
@@ -17,6 +17,8 @@ export interface QuizContentProps {
   setDescription: Dispatch<SetStateAction<CreateQuizParams['description']>>
   questions: Question[]
   setQuestions: Dispatch<SetStateAction<Question[]>>
+  category: string | null
```
[the remark this answers is not in the document]
Comment (mynotdoing) [reply src/pages/new-quiz/NewQuiz.constants.tsx:20]: fixed

**Review by Tolik170 — APPROVED** [review]

**Review by TSlashDreamy — APPROVED** [review]

File: src/constants/translations/ua/common.json [hunk 16]
```diff
@@ -37,6 +37,8 @@
   "fileSizeError": "Size for one file cannot be more than 10 MB.",
   "noItems": "Елементів не знайдено",
   "view": "Переглянути",
+  "categoryDropdown": "Категорія...",
+  "categoryDropdownHelper": "Оберіть категорію:",
```
Comment (MariaAfonina) [inline src/constants/translations/ua/common.json:41]: You don't need this anymore. Delete this.

Comment (mynotdoing) [reply src/constants/translations/ua/common.json]: fixed

**Review by MariaAfonina — APPROVED** [review]

**Pull request merged**